hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionToolNpeFix.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.regionserver;
+
+import static org.junit.Assert.assertEquals;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.HBaseClassTestRule;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.HBaseTestingUtil;
+import org.apache.hadoop.hbase.testclassification.MediumTests;
+import org.apache.hadoop.hbase.testclassification.RegionServerTests;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.util.CommonFSUtils;
+import org.apache.hadoop.util.ToolRunner;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+/**
+ * Test the issue of HBASE-28472
+ */
+@Category({ MediumTests.class, RegionServerTests.class })
+public class TestCompactionToolNpeFix extends TestCompactionTool {
```

Comment:
Do we really need to extends TestCompactionTool? Do we need to run the test in the parent class twice?

Reply:
Thanks @Apache9 for review.
In TestCompactionTool, when system is started with mslab enabled, then ChunkCreator is created and initialized. So when test case runs CompactionTool, it does not report NPE.
To reproduce the issue, setting mslab to disabled in derived class, and then base test fails with NPE as ChunkCreator is NULL.

Reply:
Then we can just change TestCompactionTool Parameterized? What is the purpose for the newly added test method?

Reply:
Other test case is added to test the CompactionTool MapReduce flow. Existing test case verifies the Non MapReduce flow.

Reply:
Then I think we should first change TestCompactionTool Parameterized, and then introduce a new tests? The current architecture seems really strange...

Reply:
Changed TestCompactionTool to Parameterized and added new cases in the same test file.